src/mon/Monitor.cc

```diff
@@ -986,6 +986,16 @@ void Monitor::shutdown()
 
   remove_all_sessions();
 
+  log_client.shutdown();
+
+  // unlock before msgr shutdown...
+  lock.Unlock();
+
+  messenger->shutdown();  // last thing!  ceph_mon.cc will delete mon.
+  mgr_messenger->shutdown();
+
+  lock.Lock();
```

Comment:
nit, why do we need to lock here?

Reply:
The release of logger in the original process is protected by lock, and it is uncertain whether concurrent access is possible after the change, so lock is added here

Reply:
could you name any of methods that could probably access `logger` with the protection of this mutex?

Reply:
Sorry.Because I'm not sure exactly what this lock needs to protect, I'd like the community members to help confirm if it's necessary to lock after the change. Now it looks like there should be only competition between Monitor::ms_dispatch() and Monitor::shutdown(), so it doesn't look like a lock is needed